source/daq/frequency_mask_trigger.cc

```diff
@@ -511,6 +555,181 @@ namespace psyllid
         }
     }
 
+
+    void frequency_mask_trigger::exe_apply_two_thresholds( exe_func_context& a_ctx )
```

Comment:
I added exe_apply_two_thresholds for the 2 threshold triggering. The first thing this method does is create 2 mask buffers from f_mask. As f_mask is created using the lower threshold, every bin of t_mask2_buffer is multiplied by f_threshold_snr_high/f_threshold_snr.